lib/Conversion/StandardToHandshake/StandardToHandshake.cpp

```diff
@@ -1341,6 +1341,45 @@ struct FuncOpLowering : public OpConversionPattern<mlir::FuncOp> {
 };
 
 namespace {
+struct DFRemoveBlockPass
+    : public PassWrapper<DFRemoveBlockPass, OperationPass<handshake::FuncOp>> {
+  void runOnOperation() override {
+    auto funcOp = getOperation();
+    auto builder = OpBuilder(funcOp.getContext());
+    auto *termOp = funcOp.getBody().front().getTerminator();
+
+    for (auto &block : funcOp) {
+      if (block.isEntryBlock())
+        continue;
+
+      // Move all operations to the first block except for terminator operations
+      // which can be safely dropped.
+      while (!block.empty()) {
```

Comment:
nit: Can you splice the operation lists instead of cloning? You can access the raw llvm::iplist via`block.getOperations()`. You can then use `splice` to move the operations over to the back of the entry block.